resources/file.go

```diff
@@ -110,6 +130,32 @@ func (obj *FileRes) Validate() error {
 	return obj.BaseRes.Validate()
 }
 
+func (obj *FileRes) mode() (os.FileMode, error) {
+	m, err := strconv.ParseInt(obj.Mode, 8, 32)
+	if err != nil {
+		return os.FileMode(0), fmt.Errorf("Mode should be an octal number (%v): %v", obj.Mode, err.Error())
+	}
+	return os.FileMode(m), nil
+}
+
+func (obj *FileRes) uid(defaultVal int) (int, error) {
+	if obj.Owner == "" {
+		return defaultVal, nil
+	}
+
+	u, err := user.Lookup(obj.Owner)
```

Comment:
Yes, we check first for username then user id, we could (and perhaps should) do the contrary.